internal/provider/kubernetes/helpers.go

```diff
@@ -214,3 +215,13 @@ func classAccepted(gc *gwapiv1.GatewayClass) bool {
 
 	return false
 }
+
+// classMarkedForDeletion returns true if the provided GatewayClass is
+// marked for deletion and its finalizer has been removed.
+func classMarkedForDeletion(gc *gwapiv1.GatewayClass) bool {
+	if gc == nil {
```

Comment:
non blocker comment:do we hit this ? I understand why its added, to make sure we dont crash below, but should `classMarkedForDeletion` return `true` when `gc` is `nil`  ?